frontend/src/js/controllers/challengeCtrl.js

```diff
@@ -63,6 +63,57 @@
 
         utilities.showLoader();
 
+        vm.toggleCollapsibility = function (rank, leaderboard) {
+            leaderboard.isCollapsed = !leaderboard.isCollapsed;
+
+            if (leaderboard.isCollapsed == false) {
+                vm.currentRank = 0;
+                angular.element('.collapsible-content_' + rank).css('height', '0px');
+                $timeout(function () {
+                    angular.element('.collapsible-content_' + rank).css('position', 'absolute');
+                    angular.element('.collapsible-content_' + rank).css('left', '-999em');
+                }, 1000);
+            } else {
+                angular.element('.collapsible-content_' + rank).css('height', '180px');
+                $timeout(function () {
+                    angular.element('.collapsible-content_' + rank).css('position', '');
+                    angular.element('.collapsible-content_' + rank).css('left', '');
+                }, 50);
```

Comment:
```suggestion
```

Reply:
Let's get rid of timeout. If after @RishabhJain2018 sees the UI and feels we need a sliding effect we can add some CSS property for it.

Reply:
@galipremsagar done